README.md

```diff
@@ -1,3 +1,203 @@
+```
+cd ~/sandbox
+git clone https://github.com/ukgovdatascience/sandbox-deploy.git
+```
+
+Create a virtualenv:
+```
+mkvirtualenv --python=/usr/local/bin/python3 sandbox-deploy
+```
+Edit ~/.virtualenvs/sandbox-deploy/bin/postactivate so it contains:
+```
+cd ~/sandbox/sandbox-deploy
+export FLASK_APP=deploy/deploy.py
+export FLASK_DEBUG=true
+export SANDBOX_DEPLOY_USERNAME=
+```
+Install the dependencies etc:
+```
+workon sandbox-deploy
+cd ..
+git clone
+pip install -r requirements.txt
+```
+
+### Use this repo
+
+Always activate the environment:
+```
+workon sandbox-deploy
+```
+
+The deploy script can be run on its own:
+```
+USERNAME=davidread deploy/deploy.sh
+```
+
+python deploy.py  #etc
+
+# examples
+python3 deploy/commands.py deploy
+python3 deploy/commands.py deploy --username=davidread --email='david.read@<domain>.gov.uk' --fullname='D Read'
+
+# run deploy app
+workon sandbox-deploy
+
+flask run --port 7000
+
+
+## Deployment to EC2
+
+```
+# join GDS vpn
+ssh -i "~/.ssh/dread-data-science-aws.pem" ubuntu@ec2-35-176-150-167.eu-west-2.compute.amazonaws.com
+git clone https://github.com/ukgovdatascience/sandbox-deploy.git
+git clone https://github.com/ukgovdatascience/data-science-sandbox-infrastucture.git
+sudo snap install kubectl --classic
+# 2017-08-24T17:29:52Z INFO cannot auto connect core:core-support-plug to core:core-support:
+# (slot auto-connection), existing connection state "core:core-support-plug core:core-support" in the way
+# kubectl 1.7.0 from 'canonical' installed
+mkdir ~/.kube
+# run locally:
+rsync -e "ssh -i ~/.ssh/dread-data-science-aws.pem" ~/.kube/sandbox.data-science.org.uk ubuntu@ec2-35-176-150-167.eu-west-2.compute.amazonaws.com:~/.kube/config
+rsync -e "ssh -i ~/.ssh/dread-data-science-aws.pem" ~/sandbox/data-science-sandbox-infrastucture/chart-env-config/sandbox/*.yml ubuntu@ec2-35-176-150-167.eu-west-2.compute.amazonaws.com:~/data-science-sandbox-infrastucture/chart-env-config/sandbox/
+# run on the EC2 box:
+kubectl config set-cluster default-cluster --server=https://dashboard.services.sandbox.data-science.org.uk
+DESIRED_VERSION=v2.3.0 bash -c 'curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | bash'
+# helm installed into /usr/local/bin/helm
+helm init
+helm repo add mojanalytics https://ministryofjustice.github.io/analytics-platform-helm-charts/charts/
+export PLATFORM_ENV=sandbox
+export USERNAME=davidread
+export EMAIL=david.read@<domain>.gov.uk
+export FULLNAME="D Read"
+export CHART_ENV_CONFIG=../data-science-sandbox-infrastucture/chart-env-config
+export HELM=helm
+export SANDBOX_DEPLOY_USERNAME=deploy
+export SANDBOX_DEPLOY_PASSWORD=<password>
+curl -O https://bootstrap.pypa.io/get-pip.py
+python3 get-pip.py
+rm get-pip.py
+pip install virtualenv
+virtualenv ~/venv
+. ~/venv/bin/activate
+cd ~/sandbox-deploy
+pip install -r requirements.txt
+pip install gunicorn
+
+export FLASK_APP=deploy/deploy.py
+flask run --port 7000
+
+sudo apt-get install nginx
```

Comment:
I'm confused here too: aren't we running this on PAAS? Why do we need to manually install Nginx?

Reply:
nope, EC2 - it's the simplest. We can iterate onto paas if we want when it is beneficial.